### DISCUSSION TITLE
Reproduce htaccess authentication behaviour using the vercel.json file

### DISCUSSION THREAD

**Discussion by ap705** [vercel/vercel#5318] [Help]
Hello everyone,

I'm facing a problem right now. I'm trying to reproduce the behaviour of a htaccess file with authentication (username and password to access the website), something like this: 

`AuthUserFile .htpasswd
AuthName "name"
AuthType Basic
require valid-user`

So I'm trying to reproduce something like this, but using the vercel.json file.

I couldn't find an example on how to do it (if it's even possible ? I don't know).

I saw that there is a paid option (150€ / month ?) to implement a "before page authentication" but I don't need something complex like that.

Thank you !

**Comment by jeffceriello** [comment]
Hi @ap705, did you find a solution to this? I also need to do something similar and the €150/month seems too expensive for what I need.

Thanks
